scripts/validateAccessibilitySpecs.js

```diff
@@ -31,50 +31,35 @@ const getAllSitePaths = (index) => {
     addUrlPaths(section.pages, startPath);
   });
 
-  return pagePaths;
+  return pagePaths.map((path) =>
+    // join path parts with __ instead of /, replace spaces with underscores
+    path.join('__').toLowerCase().replace(/\s+/g, '_'),
+  );
 };
 
 async function validate() {
-  // a map of all the pages on the apge
-  const uniqueFlatPages = {};
-
   // get a list of all of the site paths
-  const listOfPaths = getAllSitePaths(siteIndex.default);
-
-  listOfPaths.forEach((path) => {
-    // we use the page name as the final path
-    const lastNode = path.pop();
-
-    // if there's a collision (e.g. web/avatar and iOS/Avatar)
-    if (uniqueFlatPages[lastNode]) {
-      // require the spec file name to be a level more specific. e.g. avatar_ios, avatar_web
-      if (uniqueFlatPages[lastNode] !== 'collision') {
-        const existingPath = uniqueFlatPages[lastNode];
-        uniqueFlatPages[`${lastNode}_${existingPath.pop()}`] = existingPath;
-      }
-
-      uniqueFlatPages[`${lastNode}_${path.pop()}`] = path;
-      uniqueFlatPages[lastNode] = 'collision';
-    } else {
-      uniqueFlatPages[lastNode] = path;
-    }
-  });
-
-  const pages = Object.keys(uniqueFlatPages)
-    .filter((key) => uniqueFlatPages[key] !== 'collision')
-    .map((page) => page.toLocaleLowerCase());
-
-  const a11ySpecFiles = (await globby([specFile])).map((file) => file.toLocaleLowerCase());
-
-  const pagesWithoutA11ySpecFiles = pages.filter(
-    (page) => !a11ySpecFiles.includes(specFile.replace('*', page.toLocaleLowerCase())),
+  const routes = getRoutes(siteIndex.default);
+
+  const a11ySpecFiles = (await globby([specFileGlob])).map((file) =>
+    file
+      .toLowerCase()
+      // remove the playwright/accessibility/ prefix
+      .replace(/^playwright\/accessibility\//, '')
+      // remove the .spec.mjs suffix
+      .replace(/\.spec\.mjs$/, ''),
   );
 
+  const pagesWithoutA11ySpecFiles = routes.filter((page) => !a11ySpecFiles.includes(page));
```

Comment:
The consistent format makes this check much simpler — convert the route path from the index to match the file name format and do a simple comparison